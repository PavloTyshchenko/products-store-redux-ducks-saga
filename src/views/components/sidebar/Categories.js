import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { ListGroup } from 'react-bootstrap';

import * as productDuck from '../../../ducks/products';

import './Categories.css';

const Categories = (props) => {

    const { 
        history,
        categories,
        category,
        search_term,
        searchProducts
     } = props;

    const items = categories.map((item, key) => {

        return (
            <ListGroup.Item
                active={item === category}
                key={key}
                as="li"
                onClick={() => onItemSelect(item)}>
                {item}
            </ListGroup.Item>
        );
    });

    const onItemSelect = (item) => {
        if (search_term) {
            history.push(`/products/${item}/${search_term}`);
        } else {
            history.push(`/products/${item}`);
        }
        searchProducts(item, '');
    };

    return (
        <ListGroup as="ul" className="mb-3">
            {items}
        </ListGroup>
    );
};

const mapStateToProps = (state) => {
    return {
        categories: state.products.categories,
        category: state.products.category,
        search_term: state.products.search_term
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        searchProducts: (category, search_term) => {
            dispatch(productDuck.searchProducts(category, search_term));
        }
    };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Categories));
