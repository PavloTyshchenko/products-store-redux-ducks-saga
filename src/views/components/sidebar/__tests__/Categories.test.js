import React from 'react';
import { shallow } from 'enzyme';

import { Categories } from '../Categories';

const defaultProps = {
    history: '',
    categories: ['All'],
    category: 'All',
    search_term: '',
    searchProducts: jest.fn()
};

const setup = (propsOption) => {
    const props = Object.assign(defaultProps, propsOption);
    const wrapper = shallow(<Categories {...props} />);
    return {
        wrapper,
        props
    };
};

describe('Categories Component', () => {

    it('Should render without errors', () => {
        const { wrapper } = setup();
        const component = wrapper.find(`[data-test='categoriesComponent']`);
        expect(component.length).toBe(1);
    });

    test('Should match snapshot', () => {
        const { wrapper } = setup();
        expect(wrapper.debug()).toMatchSnapshot();
    });

});