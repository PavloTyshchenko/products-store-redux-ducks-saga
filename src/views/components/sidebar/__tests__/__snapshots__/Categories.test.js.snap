// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Component Should match snapshot 1`] = `
"<ForwardRef as=\\"ul\\" className=\\"mb-3\\" data-test=\\"categoriesComponent\\">
  <Bootstrap(ListGroupItem) active={true} as=\\"li\\" onClick={[Function: onClick]}>
    All
  </Bootstrap(ListGroupItem)>
</ForwardRef>"
`;
